### DISCUSSION TITLE
Package conflicts with an existing package by the same name

### DISCUSSION THREAD

**Discussion by Xiupan** [kolbasa/cordova-plugin-apkupdater#41] [Q&A] [answered]
So it downloads the APK and attempts to install it by prompting a pop-up window on the device asking the User to Install or Cancel. When I tap Install, it says `App not installed.` `The package conflicts with an existing package by the same name.`. Any way around this? Or do I have to like change the package name every single update? Any other solutions?

**Comment by kolbasa** [comment]
My guess is that the signature of the installed app is different from the update apk, e.g. when you try to update a debug app with a release app.
The package name should of course remain the same, you should not change it between updates.

**Comment by kolbasa** [comment, marked as answer]
You can get a complete error message if you install the update apk via adb like so:
`adb install update.apk`

**Reply by javico-dev11** [reply]
Hello good evening, I have a problem similar to this:
![Screenshot_1](https://user-images.githubusercontent.com/77119874/150924810-7e75861e-8455-4954-be7c-1519a281c5c4.png)
I already compared the package id in both apk and they are the same, only the version number changes, could you guide me please.
Thank you for your help.

**Reply by kolbasa** [reply]
Hi, I am happy to help with any problems.
At first glance I would say that the generated apk is corrupted. A file size of 0.5MB is suspiciously small for me.
The plugin cannot read the apk package information.
I would try to install the update.apk manually first, e.g. via adb with `adb install update.apk`.

**Reply by javico-dev11** [reply]
Thank you for your reply, 
I tell you: 
The apk has the following size
![Screenshot_20220125_060613_com.dropbox.android.jpg](https://user-images.githubusercontent.com/77119874/150965893-22492bf1-2c41-46a5-9f8f-357d5e367f05.jpg)
I have also manually downloaded the apk from the server and installed it manually and if it works properly, how could I know if the file is corrupted?

**Reply by javico-dev11** [reply]
Why wouldn't the full file be downloaded?

**Reply by kolbasa** [reply]
Ok, next I would then check how the file is delivered from the server. Can you download the full apk file in a web browser? 
Can you give some details how you download the update? Something is not right with the download.

I can see from the log that the `Content-Length` of the file is not set on the server side. This shouldn't really matter, but is the reason why the `progress` property isn't updated, in case you're wondering.

**Reply by javico-dev11** [reply]
Yes, If I can completely download the file from any browser, I attach the URL
Url: https://www.dropbox.com/s/ubfk2r35w21rtiu/update.apk

Maybe it's because the apk is hosted on dropbox?

**Reply by kolbasa** [reply]
Yes, that is also exactly the problem. This is not a direct link to the apk. The plugin downloads the HTML page. That's why it's only 0.5MB.

The direct link is then generated by the Dropbox api and looks like this, for example:
```
https://uc18be908fdb1ee36795d7eab714.dl.dropboxusercontent.com/cd/0/get/BecQxkiJ6GsEL7P2l7Z0Rd7Bza-C0-_MpprgNbcYnUHBhp-kylj6ups3vy4FywL1xlx6be7fmQqUKZlHiFUAy3LyN--5AV7FO2XaBIeclpRW0jIbgEhcNKAC8ItS3yO8oLtfQ_zaSAXwHpzRVbVEzCKh/file?_download_id=76941087362009933673608895363145386129337091249755129769225596695
```
The link probably won't work for you. Dropbox generated this for my request.
You would need another file hoster that offers a static public link.

On Github, for example, it is possible:
```
https://raw.githubusercontent.com/kolbasa/cordova-plugin-apkupdater-demo/master/update/update.zip
```

**Reply by javico-dev11** [reply]
Ahh I understand, that makes a lot of sense, thank you very much, I'm going to try another link and I tell you. Thank you very much for your help.

**Reply by javico-dev11** [reply]
Hi @kolbasa I have changed my repository and now everything works correctly, again thank you very much for your time.

**Reply by kolbasa** [reply]
No problem. Glad to help. :)
